main/lsp/ExtractVariable.cc

```diff
@@ -467,6 +480,11 @@ MultipleOccurrenceResult VariableExtractor::getExtractMultipleOccurrenceEdits(co
     auto firstMatch = matches[0];
 
     auto whereToInsert = findWhereToInsert(*scopeToInsertIn, firstMatch);
+    if (!whereToInsert.exists()) {
+        logDebugInfo(config.logger, gs, selectionLoc,
+                     "failed to determine whereToInsert in getExtractMultipleOccurrenceEdits");
+        return {};
+    }
```

Comment:
Is it worth making `findWhereToInsert` return an `optional<Loc>` to force users to explicitly dereference the result, reminding them that they also want to look at the result?

We have a few places in `Loc.h` that use `optional` specifically for this, like `source` and `fromDetails`.

Reply:
I don't think it's worth it, so I'll leave it as is. In theory, this call shouldn't fail; if the code has gotten this far, it should complete successfully. There's also only 2 call sites, both in this file, so I don't expect any calls to accidentally use it without checking.

Reply:
> In theory, this call shouldn't fail

I made the Loc::source method return an optional specifically because "this should never crash, but in practice it's crashing a lot in ways that are tangential to normal execution, we shouldn't punish the user for our own faults"

#4075

so there is precedent for this, but it's ultimately up to you in this case.